task/buildah-oci-ta/0.4/buildah-oci-ta.yaml

```diff
@@ -697,68 +897,189 @@ spec:
       env:
         - name: BUILDAH_FORMAT
           value: $(params.BUILDAH_FORMAT)
-        - name: TASKRUN_NAME
-          value: $(context.taskRun.name)
+        - name: COMMIT_SHA
+          value: $(params.COMMIT_SHA)
+        - name: REMOVE_EXPIRES_LABEL
+          value: $(params.REMOVE_EXPIRES_LABEL)
       script: |
         #!/bin/bash
         set -e
-
         echo "[$(date --utc -Ins)] Update CA trust"
-
         ca_bundle=/mnt/trusted-ca/ca-bundle.crt
         if [ -f "$ca_bundle" ]; then
           echo "INFO: Using mounted CA bundle: $ca_bundle"
-          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          cp -vf "$ca_bundle" /etc/pki/ca-trust/source/anchors
           update-ca-trust
         fi
-
+        # Check if we're reusing an artifact
+        REUSED_IMAGE_REF=$(cat "$(results.REUSED_IMAGE_REF.path)" 2>/dev/null || echo "")
+        if [ -n "$REUSED_IMAGE_REF" ]; then
+          echo "[$(date --utc -Ins)] Reusing existing artifact - handling label removal and result emission"
+          # Use the reused image reference from the result
+          echo "Reused image reference: $REUSED_IMAGE_REF"
+          # Print event_type for debugging (Pipelines as Code variable)
+          echo "Event type: ${event_type:-not set}"
+          # Handle label removal for PR artifacts reused in push pipelines
+          # Only remove expires label if:
+          # 1. REMOVE_EXPIRES_LABEL is set to "true", OR
+          # 2. This is a push pipeline (event_type == "push") AND IMAGE_EXPIRES_AFTER is set
+          should_remove_expires=false
+          if [ "${REMOVE_EXPIRES_LABEL}" = "true" ]; then
+            echo "REMOVE_EXPIRES_LABEL is set to true, will remove expires label"
+            should_remove_expires=true
+          elif [ "${event_type:-}" = "push" ] && [ -n "${IMAGE_EXPIRES_AFTER:-}" ]; then
+            echo "This is a push pipeline (event_type=${event_type}) and IMAGE_EXPIRES_AFTER is set, will remove expires label"
+            should_remove_expires=true
+          else
+            echo "Not removing expires label: REMOVE_EXPIRES_LABEL=${REMOVE_EXPIRES_LABEL}, event_type=${event_type:-not set}, IMAGE_EXPIRES_AFTER=${IMAGE_EXPIRES_AFTER:-not set}"
+          fi
+          if [ "$should_remove_expires" = "true" ] && [ -n "${IMAGE_EXPIRES_AFTER:-}" ]; then
+            echo "[$(date --utc -Ins)] Removing expires label from reused artifact"
+            # Remove the expires label using manifest-only copy
+            if ! skopeo copy --multi-arch index-only --remove-signatures \
```

Comment:
This buildah task is single arch. While his works when copying the digest for a single-arch image, is there a reason that we need to do this here?

Reply:
this has been replaced with https://github.com/brianwcook/oci-config-mod, but manifest operations are intentionally always operating on the single-arch manifest